pkg/core/mempool/subscriptions.go

```diff
@@ -0,0 +1,78 @@
+package mempool
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+)
+
+// EventType represents mempool event type.
+type EventType byte
+
+const (
+	// TransactionAdded marks transaction addition mempool event.
+	TransactionAdded EventType = 0x01
+	// TransactionRemoved marks transaction removal mempool event.
+	TransactionRemoved EventType = 0x02
+)
+
+// Event represents one of mempool events: transaction was added or removed from mempool.
+type Event struct {
+	Type EventType
+	Tx   *transaction.Transaction
+	Data interface{}
+}
+
+// RunSubscriptions runs subscriptions goroutine if mempool subscriptions are enabled.
+// You should manually free the resources by calling StopSubscriptions on mempool shutdown.
+func (mp *Pool) RunSubscriptions() {
+	if mp.subscriptionsEnabled.Load() {
```

Comment:
I think we should panic in case of `false`. If we want to `Run*`, having it disabled is certainly either a bug or an error.

Reply:
Fixed.